xchange-core/src/test/java/org/knowm/xchange/dto/account/FundingRecordStatusTest.java

```diff
@@ -11,35 +11,35 @@
 
   @Test
   public void shouldProcessStatusDescriptionNormal() throws Exception {
-    testStatusDesc("PROCESSING", "foo", FundingRecord.Status.PROCESSING, "foo");
+    testStatusDesc(FundingRecord.Status.PROCESSING, "foo", FundingRecord.Status.PROCESSING, "foo");
   }
 
   @Test
   public void shouldProcessStatusToUpercase() throws Exception {
-    testStatusDesc("Complete", "bar", FundingRecord.Status.COMPLETE, "bar");
+    testStatusDesc(FundingRecord.Status.COMPLETE, "bar", FundingRecord.Status.COMPLETE, "bar");
   }
 
   @Test
   public void shouldProcessNullDescription() throws Exception {
-    testStatusDesc("COMPLETE", null, FundingRecord.Status.COMPLETE, null);
+    testStatusDesc(FundingRecord.Status.COMPLETE, null, FundingRecord.Status.COMPLETE, null);
   }
 
   @Test
   public void shouldProcessStatusAsDescriptionWhenDescInputNull() throws Exception {
-    testStatusDesc("Unknown", null, null, "Unknown");
+    testStatusDesc(null, null, null, null);
   }
 
   @Test
   public void shouldPrependUnrecognizedStatusStringToDescription() throws Exception {
```

Comment:
This test also doesn't seem to be refactored in a way that makes sense. Actually this seems to indicate that the new API (with enum instead of String) breaks the functionality that this test was intended to test.

Reply:
I’ll take a closer look, thank you! :)